Feat unit tests

Rashi1997 opened the pull request:
We have 49.3 coverage🎉 for voice2text!
![image](https://user-images.githubusercontent.com/22200129/100572589-d46e7980-32a3-11eb-9d28-56ea47fbaf79.png)

Rashi1997 committed:
fix: unit test for v2t

yuanningma committed:
added some unit tests

Rashi1997 committed:
feat: unit tests for SpeechRecognition

jeremykong-ur commented:
Nice. Great work Rashi and Bill!
Is there a way to get a detailed report like line numbers so that we could bump up the coverage based on that?

I believe we should aim for sth at least 80%, but I am not sure if there is anything in the code that prevents us from that.

Rashi1997 commented:
> Nice. Great work Rashi and Bill!
> Is there a way to get a detailed report like line numbers so that we could bump up the coverage based on that?
> 
> I believe we should aim for sth at least 80%, but I am not sure if there is anything in the code that prevents us from that.

I need to figure out the report thing! Let me know if someone has ideas!
Yes I was also skeptical about this! So we asked the professor today, explaining about how this is just the coverage for the api part, he said that it sounds good to him. 
Also, I think with `text2code` test cases our coverage will increase!

jeremykong-ur committed:
refactor:
1) put all the plugin related component into package edu.brown.edu.v2cplugin
2) put voice2text config files into voice2text.config
3) Change the view name from "Voice 2 Code" to "Voice2Code"
These changes should not affect any functionality. However, since some view
related components are changed, pls re-import the project and delete the
"eclipse-runtime-application" folder before running.

jeremykong-ur commented:
I got null ptr exception when running the test package as 'JUnit", I have passed all the tests though.
Is that normal?

`java.lang.NullPointerException
	at edu.brown.cs.voice2text.Microphone.open(Microphone.java:54)
	at edu.brown.cs.voice2text.tests.MicrophoneTest.openMicrophoneFalseTest(MicrophoneTest.java:66)`

jeremykong-ur committed:
add speech suggestion/adaptation feature. We can supply suggestions in edu/brown/cs/speech2code/speech2text/config/context.config

Rashi1997 committed:
fix: unit tests

jeremykong-ur committed:
Merge pull request #8 from speechtocode-brown-edu/feat-speech-adaptation

Feat: add speech suggestion/adaptation feature.

Rashi1997 committed:
Merge branch 'master' of https://github.com/speechtocode-brown-edu/Speech-Based-Code-Editor-Project into feat-unit-tests

Rashi1997 committed:
fix: config reader, createSpeechContext fix

zyyyh committed:
Create TextToCommandsTest.java

Unit tests for Text2Commands

Rashi1997 committed:
feat: added text2code from command branch

Rashi1997 committed:
fix: remove redundant plugin code

Rashi1997 committed:
fix: rename PluginButton to Voice2Code

zyyyh committed:
Merge branch 'feat-unit-tests' of https://github.com/speechtocode-brown-edu/Speech-Based-Code-Editor-Project into feat-unit-tests

Rashi1997 committed:
fix: remove text2command

zyyyh committed:
Create TextToCommandsTest.java

Unit test for commands
